### ISSUE TITLE
Unfortunate export of `backtrace_uncompress_zdebug`

### ISSUE THREAD

**Issue by GregBowyer** [rust-lang/rust#58132]
This might be related to #34984, basically making a rust `staticlib` to be consumed from a C program *which* is linked under GCC and is using the address sanitizer collides on `backtrace_uncompress_zdebug`

This error shows up in our builds (not cargo sorry) like so

```
ERROR: /home/greg/.cache/bazel/_bazel_greg/c75e2059c1e63a4c3c0afb06f978fe5a/external/rure/BUILD.bazel:30:1: Linking of rule '@rure//:rure_test' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -o bazel-out/k8-fastbuild/bin/external/rure/rure_test -Wl,--gc-sections '-fuse-ld=gold' -Wl,-no-as-needed -Wl,-z,relro,-z,now -B/usr/bin -pass-exit-codes -m64 -m64 '-fsanitize=address' ... (remaining 4 argument(s) skipped)
Use --sandbox_debug to see verbose messages from the sandbox
/usr/bin/ld.gold: error: bazel-out/k8-fastbuild/bin/external/rure/librure_impl-444955826.a(elf.o): multiple definition of 'backtrace_uncompress_zdebug'
/usr/bin/ld.gold: /usr/lib/gcc/x86_64-linux-gnu/7/libasan.a(elf.o): previous definition here
collect2: error: ld returned 1 exit status
INFO: Elapsed time: 0.837s, Critical Path: 0.28s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

## Rust version
```
greg@gregslaptop ~/work/core-ticketing-host $ ./bazel-core-ticketing-host/external/rust_linux_x86_64/bin/rustc --version
rustc 1.31.0 (abe02cefd 2018-12-04)
greg@gregslaptop ~/work/core-ticketing-host $
```

## `*backtrace*` symbols from `librure_impl.a`

```
bingrep -D -s backtrace_* bazel-out/k8-fastbuild/bin/external/rure/librure_impl-444955826.a  | grep backtrace
```

```
Symbol                            Owner
  backtrace_alloc                   alloc.o
  backtrace_free                    alloc.o
  backtrace_vector_finish           alloc.o
  backtrace_vector_grow             alloc.o
  backtrace_vector_release          alloc.o
  backtrace_full                    backtrace.o
  backtrace_dwarf_add               dwarf.o
  backtrace_pcinfo                  fileline.o
  backtrace_syminfo                 fileline.o
  backtrace_close                   posix.o
  backtrace_open                    posix.o
  backtrace_get_view                read.o
  backtrace_release_view            read.o
  backtrace_qsort                   sort.o
  backtrace_create_state            state.o
  backtrace_initialize              elf.o
  backtrace_uncompress_zdebug       elf.o
```

Rure in this case is the regex-capi from the regex crate.

Compiling with clang, or without asan solves this, maybe something is missing in hiding symbols in `libbacktrace`?

**Comment by GregBowyer** [comment]
FWIW this might be unrelated to rust, and might be bazel rust_rules problem (https://github.com/bazelbuild/rules_rust/issues/189)

**Comment by bdfshongjie** [comment]
你好

On Tue, Mar 3, 2020 at 17:55 <notifications@github.com> wrote:

Closed #58132 <https://github.com/rust-lang/rust/issues/58132>.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rust-lang/rust/issues/58132?email_source=notifications&email_token=ANJSMYOLRGTSZA7QOQLHQYLRFWYKHA5CNFSM4GUBO6LKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOXB2F6CQ#event-3094634250>,
or unsubscribe
